io/eolearn/io/local_io.py

```diff
@@ -18,39 +18,89 @@ class ExportToTiff(SaveToDisk):
     :type feature: (FeatureType, str)
     :param folder: root directory where all Geo-Tiff images will be saved
     :type folder: str
-    :param band_count: Number of bands to be added to tiff image
-    :type band_count: int
+    :param band_count: Bands to be added to tiff image. Bands are represented by an integer `band_n`, 
+    a tuple in the form `(start_band, end_band)` or a list in the form `[band_1, band_2,...,band_n]`.
+    :type band_count: int, tuple or list
+    :param time_count: Dates to be added to tiff image. Dates are represented by an integer `date_n`, 
+    a tuple in the form `(start_date, end_date)` or a list in the form `[date_1, date_2,...,date_n]`.
+    :type time_count: int, tuple or list
     :param image_dtype: Type of data to be saved into tiff image
     :type image_dtype: numpy.dtype
     :param no_data_value: Value of pixels of tiff image with no data in EOPatch
     :type no_data_value: int or float
     """
 
-    def __init__(self, feature, folder='.', *, band_count=1, image_dtype=np.uint8, no_data_value=0):
+    def __init__(self, feature, folder='.', *, band_count=1, time_count=1, image_dtype=np.uint8, no_data_value=0):
         super().__init__(folder)
 
         self.feature = self._parse_features(feature)
         self.band_count = band_count
+        self.time_count = time_count
         self.image_dtype = image_dtype
         self.no_data_value = no_data_value
 
     def execute(self, eopatch, *, filename):
 
         feature_type, feature_name = next(self.feature(eopatch))
         array = eopatch[feature_type][feature_name]
+        dates = eopatch.timestamp
+        bands = range(array.shape[-1])
 
-        if self.band_count == 1:
-            array = array[..., 0]
+        filename_list = []
+        
+        if type(self.band_count) is list: 
+            if [date for date in self.band_count if type(elem) != int]:
+                raise ValueError('Invalid format in {} list, expected integers'.format(self.band_count))
+            array_sub = array[...,np.array(self.band_count)-1]
```

Comment:
Does this mean that if a list of bands is provided, that the list has to be 1-based instead of 0-based? I know hat rasterio count is 1-based, but I find this a bit confusing from the `eo-learn` side. I would drop the `band_count` and the `date_count` and introduce `band_indices` and `date_indices`, which are more in line with `numpy` and the whole `eo-learn` workflow, so that one could specify either:
* lists of various sizes: (i.e. [0] or [3,1,10,-1])
* tuple of (start_index, end_index) (i.e. (0,2)

this reduces the number of different input cases and makes the code simpler.